library/alloc/src/collections/btree/node.rs

```diff
@@ -1713,7 +1713,7 @@ unsafe fn slice_remove<T>(slice: &mut [MaybeUninit<T>], idx: usize) -> T {
         let len = slice.len();
         debug_assert!(idx < len);
         let slice_ptr = slice.as_mut_ptr();
-        let ret = (*slice_ptr.add(idx)).assume_init_read();
+        let ret = slice_ptr.add(idx).cast::<T>().read();
```

Comment:
The two changes in this file don't seem like improvements to me... now if the type of `slice_ptr` changes they will keep compiling but in the wrong way.

Given that we are actually reading/writing, an explicit `&`/`&mut` seems better?